Objects/object.c

```diff
@@ -1581,6 +1581,109 @@ _PyObject_GetMethod(PyObject *obj, PyObject *name, PyObject **method)
     return 0;
 }
 
+int
+_PyObject_GetMethodStackRef(PyObject *obj, PyObject *name, _PyStackRef *method)
+{
+#ifdef Py_GIL_DISABLED
+    int meth_found = 0;
+
+    assert(PyStackRef_IsNull(*method));
+
+    PyTypeObject *tp = Py_TYPE(obj);
+    if (!_PyType_IsReady(tp)) {
+        if (PyType_Ready(tp) < 0) {
+            return 0;
+        }
+    }
+
+    if (tp->tp_getattro != PyObject_GenericGetAttr || !PyUnicode_CheckExact(name)) {
+        *method = PyStackRef_FromPyObjectSteal(PyObject_GetAttr(obj, name));
+        return 0;
+    }
+
+    // Directly set it to that if a GC cycle happens, the descriptor doesn't get
+    // evaporated.
+    // This is why we no longer need a strong reference for this if it's
+    // deferred.
+    _PyType_LookupStackRef(tp, name, method);
+    _PyStackRef descr_st = *method;
+    descrgetfunc f = NULL;
+    if (!PyStackRef_IsNull(descr_st)) {
+        if (_PyType_HasFeature(PyStackRef_TYPE(descr_st), Py_TPFLAGS_METHOD_DESCRIPTOR)) {
+            meth_found = 1;
+        } else {
+            f = PyStackRef_TYPE(descr_st)->tp_descr_get;
+            if (f != NULL && PyDescr_IsData(PyStackRef_AsPyObjectBorrow(descr_st))) {
+                *method = PyStackRef_FromPyObjectSteal(f(PyStackRef_AsPyObjectBorrow(descr_st), obj, (PyObject *)Py_TYPE(obj)));
+                PyStackRef_CLOSE(descr_st);
+                return 0;
+            }
+        }
+    }
+    PyObject *dict;
+    if ((tp->tp_flags & Py_TPFLAGS_INLINE_VALUES) &&
+        _PyObject_TryGetInstanceAttributeStackRef(obj, name, method)) {
+        if (!PyStackRef_IsNull(*method)) {
+            PyStackRef_XCLOSE(descr_st);
+            return 0;
+        }
+        dict = NULL;
+    }
+    else if ((tp->tp_flags & Py_TPFLAGS_MANAGED_DICT)) {
+        dict = (PyObject *)_PyObject_GetManagedDict(obj);
+    }
+    else {
+        PyObject **dictptr = _PyObject_ComputedDictPointer(obj);
+        if (dictptr != NULL) {
+            dict = *dictptr;
+        }
+        else {
+            dict = NULL;
+        }
+    }
+    if (dict != NULL) {
+        Py_INCREF(dict);
+        if (_PyDict_GetItemStackRef(dict, name, method) != 0) {
+            // found or error
+            Py_DECREF(dict);
+            PyStackRef_CLOSE(descr_st);
+            return 0;
+        }
+        // not found
+        Py_DECREF(dict);
+    }
+
+    if (meth_found) {
+        *method = descr_st;
+        return 1;
+    }
+
+    if (f != NULL) {
+        *method = PyStackRef_FromPyObjectSteal(f(PyStackRef_AsPyObjectBorrow(descr_st), obj, (PyObject *)Py_TYPE(obj)));
```

Comment:
I'd like to avoid adding any more `PyStackRef_FromPyObjectSteal()` with potentially `NULL` arguments as the `NULL` check is adding overhead to every conversion.